pkg/model/trace/combine.go

```diff
@@ -90,9 +93,17 @@ func (c *Combiner) ConsumeWithFinal(tr *tempopb.Trace, final bool) (int, error)
 				}
 			}
 		}
-		return spanCount, c.sizeError()
+		maxSizeErr := c.sizeError()
+		if maxSizeErr != nil && c.allowPartialTrace {
```

Comment:
nit: The logic here seems a little repetitive.  I might move it into `sizeError` and change the signature to `check() (done bool, error)`.  It could check allowPartialTrace, and the cached maxTraceSizeReached bool, etc.

```
if done, err := c.check(); done {
   return spanCount, err
}
```

Reply:
Makes sense to me